src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -193,23 +209,53 @@ void DlgPrefSound::slotApply() {
         return;
     }
 
+    m_config.clearInputs();
+    m_config.clearOutputs();
+    emit(writePaths(&m_config));
+
+    EngineMaster::MicMonitorMode configuredMicMonitorMode =
+        static_cast<EngineMaster::MicMonitorMode>(
+              static_cast<int>(m_pMicMonitorMode->get()));
+    if (configuredMicMonitorMode == EngineMaster::MicMonitorMode::DIRECT_MONITOR
+        && m_config.hasMicInputs()) {
+        if (m_pLatencyCompensation->get() == 0.0) {
+            // TODO(Be): Make the "User Manual" text link to the manual in this dialog.
+            QMessageBox measureRoundTripLatencyAdvisoryBox;
+            measureRoundTripLatencyAdvisoryBox.setIcon(QMessageBox::Information);
+            measureRoundTripLatencyAdvisoryBox.setText(tr("Measure Round Trip Latency"));
+            measureRoundTripLatencyAdvisoryBox.setInformativeText(tr("Your microphone input will be out of time in your recorded and broadcasted mixes compared to what you hear. To align them, measure the round trip latency through your setup and enter this value for the Microphone Latency Compensation. Refer to the Mixxx Manual for details."));
```

Comment:
Pressing the "Ok" button and the "Apply" button both call DlgPreferencePage::slotApply, so there is no way for DlgPrefSound to distinguish between them. If the user has changed settings and presses either button, the dialog will show if they have direct monitor mode configured, a microphone configured, and the Microphone Latency Compensation at 0.

> It may also happen, that the user has already entered a different context and he might be confused why this pop up appears just now.

When the Ok button was pressed, preferences changed on any page were applied before the window closed. I have changed it to be consistent with the Apply button. The Ok button only applies settings changed on the current page now. This is not really the correct solution though. The correct solution is to ask the user if they want to apply or cancel changed settings when switching from one preference page to another, not just silently discard changes from other pages. But fixing that is beyond the scope of this PR.